### PR TITLE
Add optimistic governor module

### PR DESCRIPTION
**Pull request by abg4**
### Title

*What is your module called?*
Optimistic Governor

### Description

The Optimistic Governor contract is used to govern a Gnosis Safe based on a set of rules defined off-chain. These rules are enforced using UMA's [Optimistic Oracle](https://umaproject.org/optimistic-oracle.html) and the [ZODIAC identifier](https://github.com/UMAprotocol/UMIPs/blob/master/UMIPs/umip-152.md) allowing for a new era of flexible "optimistic governance". 

The Optimistic Governor allows any address to propose transactions that follow the predetermined set of rules. These transactions then go through a challenge period where any address can dispute the transaction. 

If a transaction is not disputed, it can then be executed. However, if any address believes the proposed transaction does not follow the rules, the transaction can be disputed. A dispute is resolved by neutral human voters brought in to vote on the outcome based on the predetermined set of rules.

### Audit

https://blog.openzeppelin.com/uma-optimistic-governor-audit/

### Checklist

This PR:

- [x] Adds details of the module to the [README](../../README.md) (items are sorted alphabetically)
- [x] Adds addresses of the canonical deployments of the mastercopy for your module to [deployments.json](../../src/deployments.json), along with addresses and ABI details to [constants.ts](../../src/factory/constants.ts). Ideally these should be deterministic deployments that can be easily replicated on other supported networks.
- [x] Includes an audit from a reputable auditor.

### CODE REVIEW

**Comment by abg4** [comment]
I have read the CLA Document and I hereby sign the CLA

File: README.md [hunk 1]
```diff
@@ -62,6 +62,7 @@ contract MyModule is Module {
 #### Modules
 - **[Bridge](https://github.com/gnosis/zodiac-module-bridge)** (developed by [Gnosis Guild](https://twitter.com/gnosisguild)): This module allows an address on one chain to control an avatar on another chain using an Arbitrary Message Bridge (AMB). This enables a DAO on one chain to control assets and interact with systems like a Gnosis Safe on a different chain.
 - **[Exit](https://github.com/gnosis/zodiac-module-exit)** (developed by [Gnosis Guild](https://twitter.com/gnosisguild)): This module allows users to redeem a designated token for a relative share of an avatar's assets, similar to MolochDAO's infamous rageQuit() function. 
+- **[Optimistic Governor](https://docs.outcome.finance/optimistic-governance/what-is-the-optimistic-governor)** (developed by [Outcome Finance](https://www.outcome.finance/): This module allows on-chain executions based on Snapshot proposal results. The module utilizes UMA's optimistic oracle to govern a Gnosis Safe based on a set of rules defined off-chain.
```
Comment (auryn-macmillan) [inline README.md:65]: This list should be in alphabetical order.

Comment (abg4) [reply README.md:65]: Is it just going by the first letter? `O` for `Optimistic Governor` is between `E` and `R` but let me know if I am missing something.

Comment (auryn-macmillan) [reply README.md:65]: lol, yeah you're right.

**Review by auryn-macmillan — APPROVED** [review]

**Pull request merged**